File: quine.java
class quine {
  public static void main(String[] args) {
    String s = "class quine {\n  public static void main(String[] args) {\n    String s = \"%s\";\n    System.out.printf(s, s.replaceAll(\"\\\\\\\\\", \"\\\\\\\\\\\\\\\\\").replaceAll(\"\\n\", \"\\\\\\\\n\").replaceAll(\"\\\"\", \"\\\\\\\\\\\"\"));\n  }\n}\n";
    System.out.printf(s, s.replaceAll("\\\\", "\\\\\\\\").replaceAll("\n", "\\\\n").replaceAll("\"", "\\\\\""));
  }
}

File: quine.scala
val src = List(
  "val src = List(",
  "<Quine>",
  ")",
  "src foreach {",
  "  case \"<Quine>\" =>",
  "    println(src map(\"  \\\"\" + _.replaceAll(raw\"\\\\\", raw\"\\\\\\\\\").replaceAll(\"\\n\", raw\"\\n\").replaceAll(\"\\\"\", \"\\\\\\\\\\\"\") + '\"') mkString \",\\n\")",
  "  case s => println(s)",
  "}"
)
src foreach {
  case "<Quine>" =>
    println(src map("  \"" + _.replaceAll(raw"\\", raw"\\\\").replaceAll("\n", raw"\n").replaceAll("\"", "\\\\\"") + '"') mkString ",\n")
  case s => println(s)
}
